### PR TITLE
docs: Use proper environment variables for Windows

### PR DESCRIPTION
**Pull request by T-256**
ref: https://github.com/astral-sh/uv/issues/6397#issuecomment-2304512872

### CODE REVIEW

File: docs/concepts/cache.md [hunk 1]
```diff
@@ -104,4 +104,4 @@ uv determines the cache directory according to, in order:
 
 It is important for performance for the cache directory to be located on the same file system as the
 Python environment uv is operating on. Otherwise, uv will not be able to link files from the cache
-into the environment and will instead need to fallback to slow copy operations.
+into the environment and you will instead need to use copy mode with slow operations.
```
Comment (zanieb) [inline docs/concepts/cache.md:107]: Why this change? We're talking about uv here, e.g., "uv will instead need to fallback to slow copy operations"

File: crates/uv-cli/src/lib.rs [hunk 2]
```diff
@@ -421,7 +421,7 @@ pub enum CacheCommand {
     ///
     ///
     /// By default, the cache is stored in  `$XDG_CACHE_HOME/uv` or `$HOME/.cache/uv` on Unix and
-    /// `{FOLDERID_LocalAppData}\uv\cache` on Windows.
```
Comment (zanieb) [inline crates/uv-cli/src/lib.rs:424]: Would be you be willing to update all of the `FOLDERID_` references? I think there are some more.

You'll need to update the generated reference with `cargo dev generate-all`

File: docs/concepts/cache.md (hunk 1 above)
Comment (T-256) [reply docs/concepts/cache.md:107]: by reading `fallback`, initially I thought it will automatically do it for us, because imo it means: if some progress failed do X instead.

Comment (zanieb) [reply docs/concepts/cache.md:107]: It does / should do it automatically for you?

Comment (T-256) [reply docs/concepts/cache.md]: Then I think Warning message could be improved to mention which mode it fallbacking, in #6397 it says:
```
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
```
Instead:
```
         Cache and target directories are on different filesystems, hardlinking not supported, switched to copy mode.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
```

File: docs/concepts/cache.md [hunk 3]
```diff
@@ -104,4 +104,4 @@ uv determines the cache directory according to, in order:
 
 It is important for performance for the cache directory to be located on the same file system as the
 Python environment uv is operating on. Otherwise, uv will not be able to link files from the cache
-into the environment and will instead need to fallback to slow copy operations.
+into the environment and uv will need to explicitly use _copy_ mode which will do slow operations.
```
Comment (T-256) [inline docs/concepts/cache.md:107]:
```suggestion
into the environment and will instead need to fallback to slow copy operations.
```

**Review by zanieb — APPROVED** [review]
Thanks!

File: docs/concepts/cache.md (hunk 1 above)
Comment (zanieb) [reply docs/concepts/cache.md]: Probably worth discussion elsewhere — we have too many issues to go through right now to discuss unrelated changes out of context.
